Added Zeeves wallet integration to ZilSwap

whenafirestartstoburn opened the pull request:
Added script reference for https://zeeves.io/sdk/sdk.min.js. The source code of this library is here - https://gitlab.com/whenafirestartstoburn/zeeves-js-auth-sdk.

Demo version with all changes temporary hosted here - https://swap-demo.zeeves.io

Video of how it should work - https://drive.google.com/file/d/19ovyoWbm455P0bpUCpxWLQKYIQjPF2Et/view?usp=sharing

RavenXce commented on src/app/components/Zeeves/Zeeves.js:
```diff
@@ -0,0 +1,16 @@
+  return (
+    <Helmet>
+        {/* Zeeves SDK */}
+        <script src={`https://${ZEEVES_HOST}/sdk/sdk.min.js?ts=${new Date().getTime().toString()}`}></script>
```
Is it possible to use a static sdk library that can be properly reviewed? Ideally uploaded to npm.

whenafirestartstoburn replied:
I think yea, but let's firstly discuss how do you see it. Personally I think current solution is also okay, but we can discuss.

"zilPay" object in window, for example, is also exported from static js script, which is not immutable and not npm library - inpage.js <script> tag added by ZilPay extension dynamically when page is loaded. You don't install it from npm and it's not the part of your bundle.

![image](https://user-images.githubusercontent.com/20051090/115968299-aa6a6c00-a550-11eb-8c6e-894842c5abba.png)

What happens here is actually the same thing - sdk.min.js exports Zeeves object to window and it is being used the same way as zilPay does.

whenafirestartstoburn replied:
made Zeeves an npm library

RavenXce commented on src/app/layouts/WalletDialog/components/ConnectWallet/ConnectWallet.tsx:
```diff
@@ -65,9 +67,13 @@ const ConnectWallet: React.FC<ConnectWalletProps & React.HTMLAttributes<HTMLDivE
+          <br />Download ZilPay{" "}
           <Link rel="noopener noreferrer" target="_blank" href="https://chrome.google.com/webstore/detail/zilpay/klnaejjgbibmhlephnhpmaofohgkpgkd">here</Link>.
+          <br />
+          <br />Or try telegram Zilliqa wallet - {" "}
```
Let's use proper branding here?

e.g. "Or try Zeeves, a Telegram based wallet"

whenafirestartstoburn replied:
Done!

![image](https://user-images.githubusercontent.com/20051090/115968180-fcf75880-a54f-11eb-9073-56b26cf797bc.png)

RavenXce commented on src/core/wallet/ZilPayConnectedWallet.ts:
```diff
@@ -20,6 +20,7 @@ export class ZilPayConnectedWallet implements ConnectedWallet {
   addressInfo: WalletAccountInfo;
 
   constructor(props: ZilPayConnectProps) {
+    console.log('ZilPay props - ' + JSON.stringify(props));
```
Please remove or use `logger()`.

whenafirestartstoburn replied:
Removed console.log()

whenafirestartstoburn commented:
@RavenXce fixed some critical errors on side of our frontend, deployed new version here - https://swap-demo.zeeves.io - from all latest code, should be okay

stevenkhong commented:
@whenafirestartstoburn the main issue with directly loading a js file is the risk of malicious code injection, it is extremely hard to ensure the integrity of `zeeves.io/sdk/sdk.min.js` for our users. The js can potentially deface anything on ZilSwap, open a new screen, or run some mining script.

The main difference with ZilPay is that it is loaded from the extension, users can disable it whenever they like or by default easily. However, it will not be as straight forward to to disable a hardcoded js loading from a url. 

The recommendations are:
1. Publish to NPM so the file is fixed by release version.
2. Publish to a Chrome extension and load the SDK from there.

whenafirestartstoburn commented:
> @whenafirestartstoburn the main issue with directly loading a js file is the risk of malicious code injection, it is extremely hard to ensure the integrity of `zeeves.io/sdk/sdk.min.js` for our users. The js can potentially deface anything on ZilSwap, open a new screen, or run some mining script.
> 
> The main difference with ZilPay is that it is loaded from the extension, users can disable it whenever they like or by default easily. However, it will not be as straight forward to to disable a hardcoded js loading from a url.
> 
> The recommendations are:
> 
> 1. Publish to NPM so the file is fixed by release version.
> 2. Publish to a Chrome extension and load the SDK from there.

@stevenkhong published NPM library - https://www.npmjs.com/package/zeeves-auth-sdk-js. Added this library as a dependency for ZilSwap webapp. Removed static script import.
